src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -508,8 +511,10 @@ class Manager : public DoutPrefixProvider {
         ++entry_idx;
       }
 
-      // wait for all pending work to finish
-      waiter.async_wait(yield);
+      if (!entries.empty()) {
```

Comment:
did something change to make this check necessary?

Reply:
nothing changed. added a debug log inside the function at some point, and i saw that the function is invoked many times unnecesarily